### PR TITLE
Moving latest App Configuration code to spring boot 3 branch

### PR DESCRIPTION
**Pull request by mrm9084**
This is the code changes from the 4.0.0 release. I had to publish all the changes to make it compatible in order for the build to pass.

### CODE REVIEW

File: sdk/spring/pom.xml [hunk 1]
```diff
@@ -35,7 +35,8 @@
         <module>spring-cloud-azure-starter-appconfiguration</module>
         <module>spring-cloud-azure-starter-cosmos</module>
         <!--
-        Due to compatibility with Spring Boot 3, temporarily excluded in 'feature/spring-cloud-azure-6.x' branch.
+        Due to compatibility with Spring Boot 3, temporarily excluded in 'feature/spring-cloud-azure-6.x'
```
Comment (chenrujun) [inline sdk/spring/pom.xml:38]: nit: Avoid bring unnecessary change.

File: sdk/spring/spring-cloud-azure-actuator/pom.xml [hunk 2]
```diff
@@ -105,6 +105,18 @@
...
+    <dependency>
+       <groupId>com.azure.spring</groupId>
+       <artifactId>spring-cloud-azure-appconfiguration-config-web</artifactId>
+       <version>5.0.0-beta.1</version><!-- {x-version-update;com.azure.spring:spring-cloud-azure-appconfiguration-config-web;current} -->
+      <optional>true</optional>
+     </dependency>
+     <dependency>
+       <groupId>com.azure.spring</groupId>
+       <artifactId>spring-cloud-azure-appconfiguration-config</artifactId>
+       <version>5.0.0-beta.1</version><!-- {x-version-update;com.azure.spring:spring-cloud-azure-appconfiguration-config-web;current} -->
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-actuator/pom.xml:117]: Wrong artifact id here.

Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-actuator/pom.xml:117]: Why this artifact not exist in main branch? https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/spring-cloud-azure-actuator/pom.xml

File: sdk/spring/spring-cloud-azure-appconfiguration-config-web/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
+# Release History
+
+## 4.7.0-beta.1 (Unreleased)
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-appconfiguration-config-web/CHANGELOG.md:3]: 4.7.0 already released.

File: sdk/spring/spring-cloud-azure-appconfiguration-config-web/pom.xml [hunk 4]
```diff
@@ -0,0 +1,206 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-appconfiguration-config-web/pom.xml:3]: Use spaces instead of tab as intent.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
+# Release History
+
+## 4.7.0-beta.1 (Unreleased)
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/CHANGELOG.md:3]: Same here.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/pom.xml [hunk 6]
```diff
@@ -0,0 +1,162 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/pom.xml:3]: Same here.

File: sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationStatusException.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+class AppConfigurationStatusException extends HttpResponseException {
+
+    AppConfigurationStatusException(final String message, final HttpResponse response, final Object value) {
+        super(message, response, value);
+    }
+
+    private static final long serialVersionUID = -2388602959090868645L;
+
+
+
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationStatusException.java:17]: nit: Remove useless empty lines.

File: sdk/spring/spring-cloud-azure-feature-management-web/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
+# Release History
+
+## 4.7.0-beta.1 (Unreleased)
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-feature-management-web/CHANGELOG.md:3]: Same here.

File: sdk/spring/spring-cloud-azure-feature-management-web/README.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# Spring Cloud for Azure feature management web client library for Java
+
+See: [Spring Cloud Azure Feature Management](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/spring/spring-cloud-azure-feature-management)
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-feature-management-web/README.md:3]: How about using relative path (`../spring-cloud-azure-feature-management`)? Same to all other places.

File: sdk/spring/spring-cloud-azure-feature-management-web/pom.xml [hunk 10]
```diff
@@ -0,0 +1,114 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-feature-management-web/pom.xml:2]: Same here.

File: sdk/spring/spring-cloud-azure-feature-management/CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
+# Release History
+
+## 4.7.0-beta.1 (Unreleased)
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-feature-management/CHANGELOG.md:3]: Same here.

File: sdk/spring/spring-cloud-azure-feature-management/pom.xml [hunk 12]
```diff
@@ -0,0 +1,135 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-feature-management/pom.xml:2]: Same here.

File: sdk/spring/spring-cloud-azure-integration-test-appconfiguration-config/pom.xml [hunk 13]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+  <parent>
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-integration-test-appconfiguration-config/pom.xml:8]: Please uniform the indent in this file: Use 2 spaces or 4 spaces as indent.

File: sdk/spring/spring-cloud-azure-starter-appconfiguration-config/CHANGELOG.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
+# Release History
+
+## 4.7.0-beta.1 (Unreleased)
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-starter-appconfiguration-config/CHANGELOG.md:3]: Same here.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-actuator/pom.xml (hunk 2 above)
Comment (mrm9084) [reply sdk/spring/spring-cloud-azure-actuator/pom.xml:117]: I think this has to do with me adding module-info, for some reason it can't be found unless I explicitly say it, even though the web version has it too.

**Review by saragluna — APPROVED** [review]

**Comment by saragluna** [comment]
Have you tested building a native executable with the App Configuration library?

File: sdk/spring/spring-cloud-azure-actuator-autoconfigure/pom.xml [hunk 15]
```diff
@@ -53,6 +53,11 @@
       <artifactId>spring-cloud-azure-autoconfigure</artifactId>
       <version>5.1.0-beta.1</version> <!-- {x-version-update;com.azure.spring:spring-cloud-azure-autoconfigure;current} -->
     </dependency>
+    <dependency>
+      <groupId>com.azure.spring</groupId>
+      <artifactId>spring-cloud-azure-appconfiguration-config-web</artifactId>
+      <version>5.0.0-beta.1</version><!-- {x-version-update;com.azure.spring:spring-cloud-azure-appconfiguration-config-web;current} -->
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-actuator-autoconfigure/pom.xml:59]: Should be 5.1.0-beta.1.
